ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_adaptive_virtual_topology.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+
+from .utils import UtilsMixin
+
+
+class RouterAdaptiveVirtualTopologyMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def router_adaptive_virtual_topology(self) -> dict | None:
+        """
+        Return structured config for router path-selection (DPS)
+        """
+        if not self.shared_utils.sdwan_role:
+            return None
+
+        router_adaptive_virtual_topology = {}
+
+        role = self.shared_utils.sdwan_role
+        # TODO add a key at node level to indicate this
+        if role == "transit":
+            role = "transit region"
+
+        router_adaptive_virtual_topology["topology_role"] = role
+        router_adaptive_virtual_topology["region"] = {"name": self._wan_region["name"], "id": self._wan_region["id"]}
```

Comment:
We do not need the region, zone and site for pathfinders and thus will need a check here.

Reply:
Region, zone and site are relevant to devices that create site to site tunnels via the AutoVPN mechanism (these config serve as a way to constrain the set of paths that a device should form with other devices in the network). We thus decided  that pathfinders were thus outside this heirarchy. I think some places downstream of provisioning might be relying on this now.

Reply:
Removed in latest commit - can always add it back later.